text/0000-target-bundles.md

```diff
@@ -0,0 +1,245 @@
+    "no-default-libraries": true,
+    "custom-unwind-resume": false,                          // might make sense to merge into below
+    "eh-method": "dwarf",                                   // NEW: what EH method to use
+    "dll-storage-attrs": false,                             // NEW: should use dll storage attrs?
+    "debug-info": ["Dwarf", 4],                             // NEW: what debug info format
+    "system-abi": "C",                                      // NEW: what "system" ABI means
+    "c-abi-kind": "cabi_x86_64",                            // NEW: what C-ABI implementation to use
+
+    // Various stuff adjusting how linkage is done
+    "pre-link-args": ["-Wl,--as-needed", "-Wl,-z,noexecstack"],
+    "post-link-args": [],
+    "pre-link-objects-dll": [],
+    "pre-link-objects-exe": [],
+    "post-link-objects": [],
+    "late-link-args": [],
+    "gc-sections-args": [],                                 // NEW: how to strip sections
+    "rpath-prefix": "$ORIGIN",                              // CHANGED: to allow specifying rpath prefix, null to disable rpath altogether
+    "no-compiler-rt": false,
+    "metadata-section": ".note.rustc",                      // NEW: name of section for metadata
+    "has-frameworks": false,                                // NEW: is concept of frameworks supported?
+    "position-independent-executables": true,               // should become a plain linker argument?
+    "lib_allocation_crate": "alloc_system",
+    "exe_allocation_crate": "alloc_jemalloc",
+    // should become templates? `lib{}.so` is much nicer
+    "dll_prefix": "lib",
+    "dll_suffix": ".so",
+    "exe_suffix": "",
+    "staticlib_prefix": "lib",
+    "staticlib_suffix": ".a",
+
+    // LLVM options
+    "cpu": "x86-64",                                        // CPU of the target
+    "features": "",                                         // LLVM features
+    "relocation_model": "pic",
+    "code_model": "default",
+    "eliminate-frame-pointer": true,
+
+    // Apparently OS X is so much of a special case that even moving all the special cases into
+    // configuration is painful.
+    is_like_osx: false,
+
+    // is_like_solaris is handled by the extra gc-sections key
+    // is_like_msvc is handled by the extra metadata-section, linker-kind, eh-method,
+    // dll-storage-attrs, debug-info keys
+    // is_like_windows is handled by the extra system-abi and c-abi-kind keys
+    // is_like_android is handled by the extra debug-info key
+}
+```
+
+## Distribution of targets
+
+Currently every built-in target is distributed along with the rustc compiler. This is suboptimal,
+because in majority of cases users are interested in targets which rustc can compile for, rather
+than the built-in targets rustc knows about, therefore distributing targets built-in into rustc is
+providing no benefits.
+
+Native libraries and linkers aside, it is obvious there’s little sense in distributing targets
```

Comment:
Getting rid of precompiled `std` altogether isn't something I had considered in my other comments here. If we're sure that is going to happen (and as a result `${libdir}/rustlib/<target>/*` is going to go away as most places that isn't writable) then it definitely makes sense to distribute specs for all supported targets at once rather than requiring them to each be downloaded separately because the size of a `target` will just be the size of it's spec.